language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -174,14 +175,36 @@ impl DiemVM {
                 .map_err(|e| e.into_vm_status())?;
 
             match payload {
-                TransactionPayload::Script(script) => session.execute_script(
-                    script.code().to_vec(),
-                    script.ty_args().to_vec(),
-                    convert_txn_args(script.args()),
-                    vec![txn_data.sender()],
-                    cost_strategy,
-                    log_context,
-                ),
+                TransactionPayload::Script(script) => {
+                    let diem_version = self.0.get_diem_version()?;
+                    let remapped_script =
+                        if diem_version < diem_types::on_chain_config::DIEM_VERSION_2 {
+                            None
+                        } else {
+                            script_to_script_function::remapping(&txn_data.script_hash)
```

Comment:
Thoughts on matching on the exact bytes vs the hash here (given that both are available to us)?

Reply:
Sure why not